libweb3jsonrpc/Eth.cpp

```diff
@@ -258,24 +258,52 @@ string Eth::eth_sendTransaction(Json::Value const& _json)
 		case TransactionRepercussion::ProxySuccess:
 			return toJS(n.hash);// TODO: give back something more useful than an empty hash.
 		case TransactionRepercussion::UnknownAccount:
-			BOOST_THROW_EXCEPTION(JsonRpcException("Account unknown."));
+			throw JsonRpcException("Account unknown.");
 		case TransactionRepercussion::Locked:
-			BOOST_THROW_EXCEPTION(JsonRpcException("Account is locked."));
+			throw JsonRpcException("Account is locked.");
 		case TransactionRepercussion::Refused:
-			BOOST_THROW_EXCEPTION(JsonRpcException("Transaction rejected by user."));
+			throw JsonRpcException("Transaction rejected by user.");
 		case TransactionRepercussion::Unknown:
-			BOOST_THROW_EXCEPTION(JsonRpcException("Unknown reason."));
+			throw JsonRpcException("Unknown reason.");
```

Comment:
Here we could handle also `default` case and then we could get rid of weirdly-looking throw before `return std::string()` i the end of the funcition

Reply:
This error message if it's surfaced to the caller (I see now it's not) should better be a little more specific like "Unknown authentication error."

Reply:
Thanks, I wanted to keep my changes as scoped as possible but since I'm already touching so much of `Eth::eth_sendTransaction` I'll go ahead and handle the default case and update the error message. Even though the error isn't surfaced to the caller (yet), I think that changing it now will leave the function in a good state for whomever fixes #5090